test/gtest/common/test_perf.cc

```diff
@@ -292,20 +296,35 @@ void test_perf::run_test(const test_spec& test, unsigned flags, bool check_perf,
             UCS_TEST_MESSAGE << result_str << " (attempt " << i << ")";
         }
 
-        if (perf_value != NULL) {
-            *perf_value = value;
-        }
-
         if (!check_perf) {
-            return; /* Skip */
+            return value; /* Skip */
         } else if ((value >= test.min) && (value <= test.max)) {
-            return; /* Success */
+            return value; /* Success */
         } else {
             ucs::safe_sleep(ucs::perf_retry_interval);
         }
     }
 
-     ADD_FAILURE() << "Invalid " << test.title << " performance, expected: " <<
-                      std::setprecision(3) << test.min << ".." << test.max;
+    ADD_FAILURE() << "Invalid " << test.title << " performance, expected: "
+                  << std::setprecision(3) << test.min << ".." << test.max;
+
+    return 0.0;
 }
 
+void test_perf::test_adjust(test_spec &test)
```

Comment:
can we remove this function, and put the relevant code into test_ucp/test_uct?

Reply:
test_spec min/max are only defined for the test perf, it does make sense to move it up there.

Reply:
Adding @dmitrygx since this is his code. IMHO from OOP standpoint it does not make much sense

Reply:
addressed. Removed test adjust (I don't use it).